contrib/babelfishpg_tsql/src/pl_handler.c

```
@@ -2766,6 +2770,119 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
+						{
+							ereport(ERROR,
+									(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+										errmsg("TSQL ALTER VIEW is not supported from PostgreSQL endpoint.")));
+						}
+						break;
+					}
+
+					needCleanup = isCompleteQuery && EventTriggerBeginCompleteQuery();
+			
+					PG_TRY();
+					{
+						StartTransactionCommand();
+						
+						if (isCompleteQuery)
+							EventTriggerDDLCommandStart(parsetree);
+
+						/* Get the old view's OID and verify it exists */
+						oldViewOid = RangeVarGetRelid(stmt->view, AccessExclusiveLock, true);
+						
+						/* If the view does not exist, check if the stmt is create or alter */
+						if (!OidIsValid(oldViewOid))
+						{
+							if(stmt->replace)	/* we have set replace to false for CREATE OR ALTER VIEW to avoid view does not exist error */
+							{
+								ereport(ERROR,
+										(errcode(ERRCODE_UNDEFINED_TABLE),
+											errmsg("view \"%s\" does not exist", 
+												stmt->view->relname)));
+							}
+							/* View doesn't exist - create it */
+							address = DefineView(stmt, queryString, pstmt->stmt_location, pstmt->stmt_len);
+							CommandCounterIncrement();
+						}
+						else
+						{
+							/* Save ACL before dropping the view */
+							oldViewAcl = get_old_view_acl(oldViewOid);
+							CacheInvalidateRelcacheByRelid(oldViewOid);
+			
+							/* Drop the old view */
+							originalView.objectId = oldViewOid;
+							originalView.classId = RelationRelationId;
+							originalView.objectSubId = 0;
+							performDeletion(&originalView, DROP_RESTRICT, 0);
+							CommandCounterIncrement();
+			
+							/* Create new view */
+							stmt->replace = true; 
+							address = DefineView(stmt, queryString, pstmt->stmt_location, pstmt->stmt_len);
+							CommandCounterIncrement();
+							
+							/* Store the view definition in babelfish_view_def */
+							if (store_view_definition_hook)
+								store_view_definition_hook(queryString, address);
+							CommandCounterIncrement();
```

this should be inside the if stmt. 

or rather, should we throw an error if `babelfish_view_def` is not defined for some reason? what are the consequences if the code completes without storing the view in bbf_view_def?

---

We always need to store view in `bbf_view_def`. If that's the case should I remove if stmt? Also can I use `pltsql_store_view_definition` function instead?